### PR TITLE
Updated post/comment APIs to enable ignoring future reports

### LINKED ISSUE #427: API to ignore future content reports
We need to support this praw API so that moderators can ignore future reports: http://praw.readthedocs.io/en/latest/code_overview/other/submissionmoderation.html#praw.models.reddit.submission.SubmissionModeration.ignore_reports

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
Fixes #427 

#### What's this PR do?
Adds functionality to comment/post APIs to ignore future reports on the items

#### How should this be manually tested?
- Report a comment and a post
- Verify the reports show up on `/api/v0/channels/:channel_name/reports/`
- Go to `/api/v0/posts/:post_id` and `/api/v0/comments/:comment_id` and patch with `{"ignore_reports": true}`
- Verify the reports no longer show up on `/api/v0/channels/:channel_name/reports/`
- Report the comment and post again under a **different user**
- Verify the reports still don't show up on `/api/v0/channels/:channel_name/reports/`

### CODE REVIEW

File: channels/serializers.py [hunk 1]
```diff
@@ -262,6 +263,12 @@ def update(self, instance, validated_data):
             elif removed is False:
                 api.approve_post(post_id)
 
+        if "ignore_reports" in validated_data:
+            ignore_reports = validated_data["ignore_reports"]
+            if ignore_reports is True:
```
Comment (noisecapella) [inline channels/serializers.py]: Is there the ability to stop ignoring reports if this is false?

**Comment by noisecapella** [comment]
Just the one question, looking at functionality now...

File: channels/serializers.py (hunk 1 above)
Comment (rhysyngsun) [reply channels/serializers.py]: There is in `praw`, but there's no requirement from a product standpoint at the moment.

**Comment by noisecapella** [comment]
I'm noticing the toast message is sticking around indefinitely:
screenshot from 2018-01-22 10-46-44

**Comment by noisecapella** [comment]
To reproduce, report a post twice in a row. The second time it doesn't go away

**Pull request merged**